pkgs/development/python-modules/piano-transcription-inference/default.nix

```
@@ -0,0 +1,71 @@
+{ stdenv
+, lib
+, buildPythonPackage
+, fetchPypi
+, fetchpatch
+, fetchurl
+, librosa
+, matplotlib
+, mido
+, pytorch
+, torchlibrosa
+}:
+
+buildPythonPackage rec {
+  pname = "piano-transcription-inference";
+  version = "0.0.5";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-nbhuSkXuWrekFxwdNHaspuag+3K1cKwq90IpATBpWPY=";
+  };
+
+  checkpoint = fetchurl {
+    name = "piano-transcription-inference.pth";
+    url = "https://zenodo.org/record/4034264/files/CRNN_note_F1%3D0.9677_pedal_F1%3D0.9186.pth?download=1";
+    sha256 = "sha256-w/qXMHJb9Kdi8cFLyAzVmG6s2gGwJvWkolJc1geHYUE=";
+  };
+
+  propagatedBuildInputs = [
+    librosa
+    matplotlib
+    mido
+    pytorch
+    torchlibrosa
+  ];
+
+  patches = [
+    # Fix run against librosa 0.9.0
+    # https://github.com/qiuqiangkong/piano_transcription_inference/pull/10
+    (fetchpatch {
+      url = "https://github.com/qiuqiangkong/piano_transcription_inference/commit/b2d448916be771cd228f709c23c474942008e3e8.patch";
+      sha256 = "sha256-8O4VtFij//k3fhcbMRz4J8Iz4AdOPLkuk3UTxuCSy8U=";
+    })
+  ];
+
+  postPatch = ''
+    substituteInPlace piano_transcription_inference/inference.py --replace \
+      "checkpoint_path='{}/piano_transcription_inference_data/note_F1=0.9677_pedal_F1=0.9186.pth'.format(str(Path.home()))" \
+      "checkpoint_path='$out/share/checkpoint.pth'"
+  '';
+
+  postInstall = ''
+    mkdir "$out/share"
+    cp "${checkpoint}" "$out/share/checkpoint.pth"
+  '';
+
+  # Make NUMBA_CACHE_DIR writable, or RuntimeError: cannot cache function
+  # '__shear_dense': no locator available
+  # https://github.com/numba/numba/issues/4032#issuecomment-488102702
+  checkPhase = ''
+    export NUMBA_CACHE_DIR="$(mktemp -d)"
+  '';
```

```suggestion
  preCheck = ''
    export NUMBA_CACHE_DIR="$(mktemp -d)"
  '';
```

Or else you are overwriting the standard `checkPhase`, so I'm not sure you're running any tests right now like this.

---

The `pythonImportsCheck` need `export NUMBA_CACHE_DIR="$(mktemp -d)"`, I've already try `preCheck`, but `preCheck` does not work.

---

Ah, I didn't understand that it was only set for the imports check. Please add a comment indicating whether there are tests and that this environment is set up for the imports check.

---

Done.